app/models/role.rb

```diff
@@ -64,7 +64,7 @@ def has_permission?(perm)
   end
 
   def permission_names
-    @permission_names ||= permissions.map { |p| p.name.to_sym }
+    @permission_names ||= permissions.pluck('permissions.name').map(&:to_sym)
```

Comment:
The full name is preferred, it can be needed on Rails 3.2.8 if there happens to be a join in the query.